### ISSUE TITLE
[Error Handling] Untidy error message when AuthorizationFailed is raised

### ISSUE THREAD

**Issue by jiasli** [Azure/azure-cli#15872]
**Describe the bug**

(Found this issue in #15589.)

The error message is not in a correct format:

```
> az account management-group show --name non-existing --debug

msrest.http_logger : {"error":{"code":"AuthorizationFailed","message":"The client 'test@azuresdkteam.onmicrosoft.com' with object id '6d97229a-391f-473a-893f-f0608b592d7b' does not have authorization to perform action 'Microsoft.Management/managementGroups/read' over scope '/providers/Microsoft.Management/managementGroups/non-existing' or the scope is invalid. If access was recently granted, please refresh your credentials."}}
msrest.exceptions : (AuthorizationFailed) The client 'test@azuresdkteam.onmicrosoft.com' with object id '6d97229a-391f-473a-893f-f0608b592d7b' does not have authorization to perform action 'Microsoft.Management/managementGroups/read' over scope '/providers/Microsoft.Management/managementGroups/non-existing' or the scope is invalid. If access was recently granted, please refresh your credentials.

code: AuthorizationFailed - , The client 'test@azuresdkteam.onmicrosoft.com' with object id '6d97229a-391f-473a-893f-f0608b592d7b' does not have authorization to perform action 'Microsoft.Management/managementGroups/read' over scope '/providers/Microsoft.Management/managementGroups/non-existing' or the scope is invalid. If access was recently granted, please refresh your credentials.
^^^^^                     ^ ^
```

Note that `code:` prefix, extra dash (`-`) and comma (`,`).

The code responsible for displaying the message is

https://github.com/Azure/azure-cli/blob/c6dfb4eb02e5901d4cfcb5ee7091181097048e92/src/azure-cli-core/azure/cli/core/util.py#L174-L178

Expected output:

```
AuthorizationFailed: The client 'test@azuresdkteam.onmicrosoft.com' with object id ...
```

**Comment by yonzhan** [comment]
Error Handling

**Comment by yonzhan** [comment]
@houk-ms for awareness.

**Comment by houk-ms** [comment]
@yonzhan Will investigate and fix it.

**Comment by houk-ms** [comment]
Should be fixed in #15963, close it now.
